### PR TITLE
16 cocktailstyles

### LINKED ISSUE #16: implement style for cocktails
- implement the css styles needed for the cocktails pages
- add templating mechanism as described below
- use bulk metadata for the template path
- create the template file for cocktails, providing the pocketguide block for all

### PR DESCRIPTION
**Pull request by RemoLiechti**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #16

Test URLs:
- Before: https://main--sazerac-wheatleyvodka--hlxsites.hlx.page/cocktails/spiced-winter-punch
- After: https://16-cocktailstyles--sazerac-wheatleyvodka--hlxsites.hlx.page/cocktails/spiced-winter-punch

### CODE REVIEW

File: scripts/cocktails.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    // when the metadata define a template, retrieve it
+    const resp = await fetch(`${templatePath}.plain.html`, window.location.pathname.endsWith(`${templatePath}`) ? { cache: 'reload' } : {});
+
+    if (resp.ok) {
+      // load template content
+      const tplText = await resp.text();
+
+      // retrieve the content wrapper, our top most div and put it's cocktail content aside
+      const main = doc.querySelector('main');
+      const cocktailData = main.innerHTML;
```
Comment (RemoLiechti) [inline scripts/cocktails.js:24]: instead of serialization, we should use a faster approach.

File: scripts/cocktails.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+      // apply the template content
+      main.innerHTML = tplText;
+
+      // reapply the cocktails content
+      const cocktailDiv = main.querySelector('div.cocktail');
+      cocktailDiv.innerHTML = cocktailData;
```
Comment (RemoLiechti) [inline scripts/cocktails.js:31]: instead of serialization, we should use a faster approach.

File: styles/styles.css [hunk 3]
```diff
@@ -499,3 +499,43 @@ a.navigation::after {
...
+  width: 100px;
+  height: 100px;
+  margin: 0 20px 0 0;
+  border-radius: 10px;
+}
+
+#created-by+ul > li {
+  color: #002855;
+  font-weight: 700;
+  line-height: 1.35;
```
Comment (RemoLiechti) [inline styles/styles.css:530]: vars

File: scripts/cocktails.js (hunk 1 above)
Comment (karlpauls) [reply scripts/cocktails.js:24]: As discussed, it should work something like this:

```
diff --git a/scripts/cocktails.js b/scripts/cocktails.js
index c7b5414..813f029 100644
--- a/scripts/cocktails.js
+++ b/scripts/cocktails.js
@@ -21,14 +21,15 @@ export async function loadCocktail(doc) {
 
       // retrieve the content wrapper, our top most div and put it's cocktail content aside
       const main = doc.querySelector('main');
-      const cocktailData = main.innerHTML;
+      const cocktailData = Array.from(main.children);
 
       // apply the template content
       main.innerHTML = tplText;
 
       // reapply the cocktails content
       const cocktailDiv = main.querySelector('div.cocktail');
-      cocktailDiv.innerHTML = cocktailData;
+      cocktailDiv.textContent = '';
+      cocktailDiv.append(...cocktailData);
 
       const insertLocation = cocktailDiv.previousElementSibling;
```

File: styles/styles.css [hunk 4]
```diff
@@ -499,3 +499,43 @@ a.navigation::after {
 a.navigation:hover::after {
   transform: scaleX(1);
 }
+
```
Comment (karlpauls) [inline styles/styles.css:502]: As discussed, I think we should use the templating mechanism for styles for this - IIRC, there already is a cocktails.css

File: styles/styles.css (hunk 3 above)
Comment (karlpauls) [reply styles/styles.css:530]: Yes, please.

File: styles/styles.css [hunk 5]
```diff
@@ -499,3 +499,43 @@ a.navigation::after {
...
+  margin-bottom: 30px;
+}
+
+/* cocktail created by thumbnail */
+#ingredients, #directions, #created-by {
+  font-size: var(--heading-font-size-l);
+  color: #7f5732;
+}
+
+#created-by+ul > * > * > img {
```
Comment (karlpauls) [inline styles/styles.css:520]: I don't think you need the * selectors.

**Review by karlpauls — CHANGES_REQUESTED** [review]

**Review by karlpauls — APPROVED** [review]

**Pull request merged**